internal/monitors/filtering.go

```diff
@@ -63,15 +65,10 @@ func (mf *monitorFiltering) HasEnabledMetricInGroup(group string) bool {
 	return false
 }
 
-// HasAnyNonDefaultMetricEnabled returns true if there is any custom metric
+// HasAnyExtraMetrics returns true if there is any custom metric
 // enabled for this output instance.
-func (mf *monitorFiltering) HasAnyNonDefaultMetricEnabled() bool {
-	for _, m := range mf.EnabledMetrics() {
-		if !mf.metadata.IncludedMetrics[m] {
-			return true
-		}
-	}
-	return false
+func (mf *monitorFiltering) HasAnyExtraMetrics() bool {
+	return mf.hasExtraMetrics
```

Comment:
`mf` can be `nil` if `enableBuiltInFiltering: false`.

Reply:
See https://github.com/signalfx/signalfx-agent/pull/781/files#diff-246334eb435b0d0c6d4e4357d843a959R39